docs/contribute/api-docs-contribution/guidelines.md

```diff
@@ -0,0 +1,575 @@
+      parameters:
+        - name: userId
+          in: path
+          required: true
+          description: The unique identifier of the user
+          schema:
+            type: string
+            example: "12345"
+        - name: postId
+          in: path
+          required: true
+          description: The unique identifier of the post
+          schema:
+            type: string
+            example: "post-abc"
+```
+
+For optional variations, create separate paths:
+
+```yaml
+paths:
+  /books:
+    get:
+      summary: List all books
+  /books/{category}:
+    get:
+      summary: List books by category
+      parameters:
+        - name: category
+          in: path
+          required: true
+          description: The book category to filter by
+          schema:
+            type: string
+            example: "fiction"
+```
+:::
+
+:::{tab-item} Elasticsearch
+:sync: elasticsearch
+
+For endpoints that can accept optional parameters, define multiple paths in the `urls` array:
+
+```ts
+urls: [
+  {
+    path: "/{index}/_doc/{id}",
+    methods: ["PUT", "POST"]
+  },
+  {
+    path: "/{index}/_doc",
+    methods: ["POST"]
+  }
+]
+```
+
+**Notes:**
+- The `make transform-to-openapi` output shows these two paths as separate entities
+- The `make transform-to-openapi-for-docs` output contains only the most complex variation of the paths (and mentions the others in its description)
+:::
+
+::::
+
+## Document enum values
+
+Enums define a fixed set of allowed values for a property. Well-documented enum values help users understand the purpose of each option and when to use it.
+
+Here are some principles for documenting enum values:
+
+- **Clarify non-obvious meanings:** Document when the purpose isn't clear from the name alone
+- **Provide usage guidance:** Explain when to use each option and any performance implications
+- **Document special behavior:** Note if some values are deprecated, experimental, or have unique characteristics
+- **Keep it concise:** Use brief, focused descriptions that explain the essential differences
+
+:::{tip}
+You can skip documenting enum values that are self-explanatory (like `true`/`false`) or follow standard conventions (like HTTP status codes).
+:::
+
+### Examples
+
+::::{tab-set}
+:group: implementations
+
+:::{tab-item} OpenAPI
+:sync: openapi
+
+In OpenAPI specifications, enum descriptions are documented in schema definitions:
+
+```yaml
+components:
+  schemas:
+    SearchType:
+      type: string
+      enum:
+        - query_then_fetch
+        - dfs_query_then_fetch
+        - query_and_fetch
+      description: |
+        The search execution strategy:
+        - `query_then_fetch`: Searches across all fields using default settings
+        - `dfs_query_then_fetch`: Performs distributed frequency scoring before fetching results  
+        - `query_and_fetch`: Fetches results immediately without scoring optimization
+```
+
+For inline enums in parameters:
```

Comment:
> For inline enums in parameters:

IMO this second example is redundant since it's just two variations of how to specify an array in a yaml file.

Reply:
Agreed